.github/actions/tests/run-api-tests/action.yml

```diff
@@ -0,0 +1,37 @@
+name: Run API tests
+description: Runs the WooCommerce Core API tests and generate Allure report.
+permissions: {}
+
+inputs:
+    report-name:
+        description: Name of Allure report to be generated.
+        required: true
+
+runs:
+    using: composite
+    steps:
+        - name: Run API tests.
+          id: run-api-tests
+          working-directory: plugins/woocommerce
+          shell: bash
+          env:
+              BASE_URL: http://localhost:8086
+              USER_KEY: admin
+              USER_SECRET: password
+          run: pnpm exec playwright test --config=tests/api-core-tests/playwright.config.js
+
+        - name: Generate Test report.
+          if: success() || ( failure() && steps.run-api-tests.conclusion == 'failure' )
+          working-directory: plugins/woocommerce
+          shell: bash
+          run: pnpm exec allure generate --clean ${{ env.ALLURE_RESULTS_DIR }} --output ${{ env.ALLURE_REPORT_DIR }}
+
+        - name: Archive test report
+          if: success() || ( failure() && steps.run-api-tests.conclusion == 'failure' )
```

Comment:
> Wouldn't this and the previous steps conditions mean that the case of failure() but steps.run-api-tests.conclusion != 'failure' is not covered?
>
> And if so, would that be OK?

Yup, that's fine. 

`steps.run-api-tests.conclusion != 'failure'` means that `steps.run-api-tests.conclusion` could either be `cancelled` or `skipped`. This will cause the "Archive test report" step  to not run. And that's fine because it just meant that the API/E2E/k6 tests were cancelled or skipped anyway (i.e., no test report generated, so there's nothing to archive)